multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -97,55 +81,50 @@ func NewMemberClusterAnnounceReconciler(client client.Client, scheme *runtime.Sc
 // Reconcile implements cluster status management on the leader cluster
 func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	memberAnnounce := &multiclusterv1alpha1.MemberClusterAnnounce{}
-	err := r.Get(ctx, req.NamespacedName, memberAnnounce)
+	var err error
+	memberID := getIDFromName(req.Name)
+	err = r.Get(ctx, req.NamespacedName, memberAnnounce)
 	if err != nil {
-		if !errors.IsNotFound(err) {
-			// Cannot read the requested resource. Return error, so reconciliation will be retried.
-			return ctrl.Result{}, err
-		}
-		// If MemberClusterAnnounce is deleted, no need to process because RemoveMember must already
-		// be called.
+		// If MemberClusterAnnounce is deleted, no further processing is needed, as cleanup
+		// must have been done when the Finalizer was removed.
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
 	r.mapLock.Lock()
-	// If the member is not found in timerData, it means the member is not added to the ClusterSet or was deleted.
-	if data, ok := r.timerData[common.ClusterID(memberAnnounce.ClusterID)]; ok {
+	// If the member is not found in memberStatus, it means MemberClusterAnnounce is not created
```

Comment:
Again, I think this block (from line 94 to 119) should be moved to addMemberStatus (probably rename it to addOrUpdateMemberStatus). Could you check?

Reply:
Updated.